core/lib/services/stores.spec.ts

```diff
@@ -442,5 +444,35 @@ describe(`Stores service`, () => {
 			expect(values).toEqual([1, 2]);
 			unsubscribe();
 		});
+
+		test(`should override equals function`, () => {
+			const onChangeCalls: number[][] = [];
+			const values: number[][] = [];
+			const dirtyValue$ = writable([1]);
+			const onValueChange$ = writable((value: number[]) => {
+				onChangeCalls.push(value);
+			});
+
+			const value$ = bindableDerived(
+				onValueChange$,
+				[dirtyValue$],
+				([dirtyValue]) => dirtyValue.map((dv) => Math.floor(dv)),
+				(a, b) => a.every((val, index) => val === b[index])
+			);
+			value$.subscribe((value) => values.push(value));
+			expect(values).toEqual([[1]]);
+
+			dirtyValue$.set([1]); // no change
+			expect(onChangeCalls).toEqual([]);
+			expect(values).toEqual([[1]]);
+
+			dirtyValue$.set([2.5]);
+			expect(onChangeCalls).toEqual([[2]]);
+			expect(values).toEqual([[1], [2]]);
```

Comment:
Maybe you could also check that the value of `dirtyValue$` changed as well (and is now `[2]`), and also at the end of the test (for `[5,7]`).